components/LessonTabs/LessonTabs.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import { CURRICULUM_PATH } from '../../constants'
+import NavCard from '../NavCard'
+
+type Lesson = {
+  title: string
+  docUrl?: string | null
+  slug: string
+}
+
+export enum LessonTab {
+  LESSON = 'lesson',
+  CHALLENGES = 'challenges',
+  EXERCISES = 'exercises',
+  MENTOR_EXERCISES = 'mentor exercises'
+}
+
+type Props = {
+  lesson: Lesson
+  activeTab: LessonTab
+}
+
+const LessonTabs = ({ lesson, activeTab }: Props) => {
+  const tabs = [
+    ...(lesson.docUrl ? [{ text: 'lesson', url: lesson.docUrl }] : []),
```

Comment:
We can simplify this to 

```
...(lesson.docUrl && [{ text: 'lesson', url: lesson.docUrl }])
```

Reply:
When TypeScript sees this, it throws the following:

```
Type '"" | { text: string; url: string; }[] | null | undefined' must have a '[Symbol.iterator]()' method that returns an iterator.
```

It's saying it's possible for `[{ text: 'lesson', url: lesson.docUrl }]` to be falsey and `lesson.docUrl` which has the types `string | undefined | null` to be returned. Type `string`, `undefined`, and `null` aren't iterable; therefore, it throws an error when destructuring whatever the expression returns.